File: lab_harmony.cpp
#include <iostream>
#include <tuple>
#include <vector>
using namespace std;

int harmony( int a, int b )
{
    return 0;
}





















int main()
{
    vector< tuple<int,int,int> > tests {make_tuple(42, 69, 25), 
                                        make_tuple(123, 999, 50),
                                        make_tuple(255, 254, 88),
                                        make_tuple(127, 128, 0),
                                        make_tuple(255, 126, 75),
                                        make_tuple(515, 3, 100),
                                        make_tuple(2147483647, 255, 100) };

    int errors = 0;
    for( auto test : tests )
    {
        int yourAnswer = harmony( get<0>(test), get<1>(test) );
        
        if( yourAnswer != get<2>(test) )
        {
            cout << "Error when comparing " << get<0>(test) << " and " << get<1>(test) << endl
                << "Yours  : " << yourAnswer << endl
                << "Correct: " << get<2>(test) << endl
                << endl;
            errors += 1;
        }
    }

    if( errors == 0)
    {
        cout << "Congratulations, no errors" << endl;
    }
    else
    {
        cout << "Uh oh, " << errors << " error/s remain" << endl;
    }

    return errors;
}

File: lab_harmony.py
import sys

def harmony( a, b ):
    # convert to binary
    aBinary = [int(x) for x in bin(a)[2:]]
    bBinary = [int(x) for x in bin(b)[2:]]

    # pad/trim lists to length of 8
    aBinary = [0]*8 + aBinary
    bBinary = [0]*8 + bBinary

    aBinary = aBinary[-8:]
    bBinary = bBinary[-8:]

    # how many bits match
    count = 0
    for i in range(8):
        if aBinary[i] == bBinary[i]:
            count += 1

    # figure out percentage
    return int(round(count / 8 * 100,0))






















def main():
    tests = [(42, 69, 0), \
            (123, 999, 50), \
            (255, 254, 88), \
            (127, 128, 0), \
            (255, 126, 75), \
            (515, 3, 100), \
            (2147483647, 255, 100)]

    errors = 0
    for a, b, correct in tests:
        yourAnswer = harmony( a, b )
        
        if yourAnswer != correct:
            print('Error when comparing {} and {}'.format(a, b))
            print('Yours  : {}'.format(yourAnswer))
            print('Correct: {}'.format(correct))
            print()
            errors += 1

    if errors == 0:
        print('Congratulations, no errors')
        
    else:
        print('Uh oh, {} error/s remain'.format(errors))

    return errors

if __name__ == '__main__':
    sys.exit(main())